ui/admin/app/templates/scopes/scope/host-catalogs/host-catalog/host-sets/host-set/hosts/index.hbs

```diff
@@ -28,26 +28,26 @@
 
   <page.body>
     {{#if @model.hosts}}
-      <Rose::Table as |table|>
-        <table.header as |header|>
-          <header.row as |row|>
+
+      <Hds::Table @model={{@model.hosts}}>
```

Comment:
Why we are not using table attributes `@model` and `@columns` as we did with the other HDS::Table migration?

Reply:
@calcaide because we are doing conditionals to provide different header cell data

Reply:
@dvonanderson 

Except when using `Hds::Table` for building components, in our templates I see no use of `Hds::Table` without `@columns` attribute.

The pattern I see when building table with dynamic content is checking for the condition once and rendering a table per condition. You can see this pattern in our codebase [here](https://github.com/hashicorp/boundary-ui/blob/main/ui/admin/app/templates/scopes/scope/auth-methods/auth-method/accounts/index.hbs#L27-L124), [here](https://github.com/hashicorp/boundary-ui/blob/main/ui/admin/app/templates/scopes/scope/auth-methods/auth-method/managed-groups/index.hbs#L26-L108) and [here](https://github.com/hashicorp/boundary-ui/blob/main/ui/admin/app/templates/scopes/scope/auth-methods/auth-method/managed-groups/managed-group/members.hbs#L38-L104).

With this implementation, you are evaluating `@model.hostSet.isStatic` 2 times for the header + n times for the host which is a higher computation cost compared to one evaluation.

If the condition is ONLY to provide different headers, you can evaluate once and provide different headers. I.e:
```
<Hds::Table>
    {{#if @model.hostSet.isStatic}}
        <:head>
            head content for condition A
        </:head>
    {{else}}
        <:head>
            head content for condition A
        </:head>
    {{/if}}
```

But I do think. In this specific case, does the condition JUST affects the header or does also affect the body of the table? if so, I will suggest to follow one of the approaches we already have in our codebase provided above.

Let me know your thoughts.

Reply:
@dvonanderson 

On my previous comment I was wrong: `With this implementation, you are evaluating @model.hostSet.isStatic 2 times for the header + n times for the host which is a higher computation cost compared to one evaluation` the body will keep computing that condition N times.

I like this solution. I think is worth it to bring to tech time the table patterns we are encountering. It will be good to have team consensus.

Reply:
@calcaide such a good point. Really need to start considering the computation complexity of some of the blocks that have significant conditional checks!

Reply:
@calcaide after talking with the team and researching how other teams do conditional rendering in table headers, I am going to stick with its current iteration, but I am going to clean it up a bit.  I created an controller that will handle the translations and return the array of headers thats needed to make it look more readable.